### DISCUSSION TITLE
Switch to structured layout?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-m2e/m2e-core#746] [Ideas]
Currently m2e uses a "flat" layout, that means all plugins, features and tests are in the root of the code repository tree.

Given that m2e grows over time (e.g. we want to split the archetype part, adopt certain connectors, ...) I'd like to discuss if we maybe should switch to a "structured" layout using e.g. a sub-folder for

1. plugins
2. tests
3. features
4. sites

this would also allow less configuration with the pomless build as pomless can detect new modules automatically.

**Comment by mickaelistria** [comment]
This will require to change also some build and publish scripts. I think the effort is not worth the benefit.

**Reply by HannesWell** [reply]
That's right, but I think that's doable. The m2e-setup has to be adjusted as well.

You are referring to the JenkinsFile and the Jenkins-Jobs to promote M2E (btw. could those also be defined in a file that is committed into git?)? Did I oversaw something?

**Comment by HannesWell** [comment]
That's a suggestion I fully support.

As discussed in the context of embedding the archetype connector we could also consider to have multiple `plugin/bundle` folders. Maybe `plugins/bundles-core` and `plugins/bundle-connectors`. Or the other way round. Or just `plugins/bundles` and `connectors`. IIRC folder names supported by Tycho-pomless can be configured (in the .mvn/extensions.xml?). Because m2e is only build standalone (unlike for example platform TLPs) we can effortless configure that.

Besides that I think instead of `site` we should use `releng`. It is more general and 'sites' are deprecated.

**Reply by HannesWell** [reply]
Additionally we have the `m2e-maven-runtime` and `m2e-core-tests` subfolders that IMHO should stay respectivly have to stay as they are.

**Comment by akurtakov** [comment]
I support everything that makes things more convention-over-configuration with the usual requirement that if someone starts it he is supposed to fix the releng and etc. parts too.
